zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -193,14 +221,38 @@ impl<F: Field> ConstraintBuilder<F> {
         });
 
         self.require_zero("initial Stack value is 0", q.initial_value.clone());
+
+        self.require_equal(
+            "state_root is unchanged for Stack",
+            q.state_root(),
+            q.state_root_prev(),
+        );
     }
 
     fn build_account_storage_constraints(&mut self, q: &Queries<F>) {
         // TODO: cold VS warm
         self.require_zero("field_tag is 0 for AccountStorage", q.field_tag());
 
-        // TODO: add mpt lookup for committed value and final value in an access
-        // group.
+        self.condition(q.last_access(), |cb| {
+            cb.add_lookup(
+                "mpt_update exists in mpt circuit for AccountStorage last access",
+                vec![
+                    (
+                        q.rw_table.address.clone(),
+                        q.mpt_update_table.address.clone(),
+                    ),
+                    (
+                        q.rw_table.storage_key.clone(),
+                        q.mpt_update_table.storage_key.clone(),
+                    ),
+                    (q.field_tag(), q.mpt_update_table.field_tag.clone()),
```

Comment:
There was a recent change in the MPT lookup [specs](https://github.com/privacy-scaling-explorations/zkevm-specs/commit/a1457dc02da26797b1f25aed0197122898761bfc) to use MPT proof type instead of field tag.

Reply:
I had in mind adding these new types that you mention, but I agree this might be better to add in a new PR.